src/components/editor-page/document-bar/revisions/revision-viewer.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { Fragment } from 'react'
+import ReactDiffViewer, { DiffMethod } from 'react-diff-viewer'
+import { useAsync } from 'react-use'
+import type { RevisionDetails, RevisionMetadata } from '../../../../api/revisions/types'
+import { getRevision } from '../../../../api/revisions'
+import { useApplicationState } from '../../../../hooks/common/use-application-state'
+import { ShowIf } from '../../../common/show-if/show-if'
+import { WaitSpinner } from '../../../common/wait-spinner/wait-spinner'
+import { useNoteMarkdownContent } from '../../../../hooks/common/use-note-markdown-content'
+import { useIsDarkModeActivated } from '../../../../hooks/common/use-is-dark-mode-activated'
+import type { AsyncState } from 'react-use/lib/useAsyncFn'
+
+export interface RevisionViewerProps {
+  selectedRevisionId?: number
+  allRevisions?: RevisionMetadata[]
+}
+
+/**
+ * Renders the diff viewer for a given revision and its previous one.
+ * @param selectedRevisionId The id of the currently selected revision.
+ * @param allRevisions List of metadata for all available revisions.
+ */
+export const RevisionViewer: React.FC<RevisionViewerProps> = ({ selectedRevisionId, allRevisions }) => {
+  const noteIdentifier = useApplicationState((state) => state.noteDetails.primaryAddress)
+  const markdownContent = useNoteMarkdownContent()
+  const darkModeEnabled = useIsDarkModeActivated()
+
+  const previousRevisionContent = useAsync(async () => {
+    if (!allRevisions || selectedRevisionId === undefined) {
+      return Promise.reject()
+    }
+    const revisionIds = allRevisions.map((revisionMetadata) => revisionMetadata.id)
+    const largestId = Math.max(...revisionIds)
+    if (selectedRevisionId === largestId) {
+      return Promise.resolve(markdownContent)
+    }
+    const nextSmallerId = revisionIds
+      .sort()
+      .reverse()
+      .find((id) => id < selectedRevisionId)
+    if (!nextSmallerId) {
+      return Promise.resolve('')
+    }
+    const revision = await getRevision(noteIdentifier, nextSmallerId)
+    return revision.content
```

Comment:
I thought the backend api should deliver a diff directly?

Reply:
It does. But our RevisionViewer does not accept diffs as argument. That's the reason, why we wanted to rebuild it on our own. But not in this PR.

Reply:
Then please add a todo

Reply:
https://github.com/hedgedoc/react-client/issues/1989